packages/dev/core/src/Misc/snapshotRenderingHelper.ts

```diff
@@ -134,6 +141,29 @@ export class SnapshotRenderingHelper {
             return;
         }
 
+        if (this._disableRenderingRefCount === 0) {
+            // Snapshot rendering switches from enabled to disabled
+            // We reset the performance priority mode to that which it was before enabling snapshot rendering, but first set it to “BackwardCompatible” to allow the system to regenerate resources that may have been optimized for snapshot rendering.
+            // We'll then restore the original mode at the next frame.
+            this._scene.performancePriority = ScenePerformancePriority.BackwardCompatible;
+            if (this._currentPerformancePriorityMode !== ScenePerformancePriority.BackwardCompatible) {
+                this._pendingCurrentPerformancePriorityMode = this._currentPerformancePriorityMode;

```

Comment:
Seems like `pendingCurrentPerformancePriorityMode` can just be a local variable. It is set to undefined in the same place that disableREnderingRefCount is set to 0, and the code already checks the value of disableRenderingRefCount. In other words, pendingCurrentPerformancePriorityMode will be undefined iff disableRenderingRefCount is 0. Is that right, or did I miss something in the logic?

Reply:
I need this to be a class variable because I reuse it here in `enableSnapshotRendering()`:

```typescript
this._currentPerformancePriorityMode = this._pendingCurrentPerformancePriorityMode ?? this._scene.performancePriority;
```
I can't use only `this._scene.performancePriority` because the value will have been set to `Backward compatible` early in `disableSnapshotRendering()`, and if we don't reach the code that sets `this._scene.performancePriority` to the right value (because snapshot mode has been reenabled before we could reach this point - line 157), we need to set the right value to `this._currentPerformancePriorityMode` and not `Backward compatible`.

Reply:
Ah ok. I'm finding it a little hard to follow the logic, but I don't have ideas on how to simplify it.